src/gui/core/gui_definition.hpp

```diff
@@ -0,0 +1,132 @@
+/*
+   Copyright (C) 2008 - 2017 by Mark de Wever <koraq@xs4all.nl>
+   Part of the Battle for Wesnoth Project http://www.wesnoth.org/
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY.
+
+   See the COPYING file for more details.
+*/
+
+#pragma once
+
+#include "config.hpp"
+
+#include "config.hpp"
+#include "gui/core/window_builder.hpp"
+#include "gui/core/widget_definition.hpp"
+
+namespace gui2
+{
+/**
+ * A GUI definiton.
+ *
+ * Each GUI contains several widgets, their definitons, and windows and controls the appearance and
```

Comment:
I don't really like that you use the term "GUI" here. Something like "theme" would be much better.